conda/core/python_dist.py

```diff
@@ -0,0 +1,444 @@
+        if fname.endswith('.dist-info'):
+            anchor_file = "%s/%s/%s" % (site_packages_dir, fname, 'RECORD')
+        elif fname.endswith(".egg-info"):
+            if isfile(join(site_packages_path, fname)):
+                anchor_file = "%s/%s" % (site_packages_dir, fname)
+            else:
+                anchor_file = "%s/%s/%s" % (site_packages_dir, fname, "PKG-INFO")
+        elif fname.endswith('.egg-link'):
+            anchor_file = "%s/%s" % (site_packages_dir, fname)
+        elif fname.endswith('.pth'):
+            continue
+        else:
+            continue
+        site_packages_anchor_files.add(anchor_file)
+
+    return site_packages_anchor_files
+
+
+def get_dist_file_from_egg_link(egg_link_file, prefix_path):
+    """"""
+    egg_info_full_path = None
+
+    with open(join(prefix_path, win_path_ok(egg_link_file))) as fh:
+        egg_link_contents = fh.readlines()[0].strip()
+
+    egg_info_fnames = glob(join(egg_link_contents, "*.egg-info"))
+    if egg_info_fnames:
+        assert len(egg_info_fnames) == 1, (egg_link_file, egg_info_fnames)
+        egg_info_full_path = join(egg_link_contents, egg_info_fnames[0])
+
+        if isdir(egg_info_full_path):
+            egg_info_full_path = join(egg_info_full_path, "PKG-INFO")
+
+    return egg_info_full_path
+
+
+def get_python_distribution_info(anchor_file, prefix_path):
+    """"""
+    if anchor_file.endswith('.egg-link'):
+        sp_reference = None
+        dist_file = get_dist_file_from_egg_link(anchor_file, prefix_path)
+        dist_cls = PythonEggInfoDistribution
+        package_type = PackageType.SHADOW_PYTHON_EGG_LINK
+    elif ".dist-info" in anchor_file:
+        sp_reference = basename(dirname(anchor_file))
+        dist_file = join(prefix_path, win_path_ok(dirname(anchor_file)))
+        dist_cls = PythonInstalledDistribution
+        package_type = PackageType.SHADOW_PYTHON_DIST_INFO
+    elif anchor_file.endswith(".egg-info"):
+        sp_reference = basename(anchor_file)
+        dist_file = join(prefix_path, win_path_ok(anchor_file))
+        dist_cls = PythonEggInfoDistribution
+        package_type = PackageType.SHADOW_PYTHON_EGG_INFO_FILE
+    elif ".egg-info" in anchor_file:
+        sp_reference = basename(dirname(anchor_file))
+        dist_file = join(prefix_path, win_path_ok(dirname(anchor_file)))
+        dist_cls = PythonEggInfoDistribution
+        package_type = PackageType.SHADOW_PYTHON_EGG_INFO_DIR
+    else:
+        raise NotImplementedError()
+
+    try:
+        pydist = dist_cls(dist_file)
+    except MetadataConflictError:
+        print("MetadataConflictError:", anchor_file)
+        pydist = None
+
+    return pydist, sp_reference, package_type
+
+
+def get_python_record(anchor_file, prefix_path):
+    """"""
+    pydist, sp_reference, package_type = get_python_distribution_info(anchor_file, prefix_path)
+
+    if pydist is None:
+        return None
+
+    # x.provides  =>  [u'skdata (0.0.4)']
```

Comment:
Also I should erase this is no longer true :-p as we are parsing all ourselves

Reply:
ok!